### PR TITLE
Feature itunes psc rel support

### PR DESCRIPTION
**Pull request by argl**
I extended Feedme a bit and added support for the itunes namespace, podlove simple chapters and rel elements, These are all used in Podcast feeds. My Elixir skills are still a bit meh, so feel free to comment or reject. Tests and fitting fixtures are included.

### CODE REVIEW

File: test/feedme_test.exs [hunk 1]
```diff
@@ -10,24 +10,38 @@ defmodule FeedmeTest do
...
+      {:ok, _feed} = Feedme.parse(atom)
       assert called Feedme.Parsers.Atom.valid?(atom_doc)
       assert called Feedme.Parsers.Atom.parse(atom_doc)
     end
 
-    {:error, "Feed format not valid"} = Feedme.parse(wrong)
+    assert {:error, "Feed format not valid"} = Feedme.parse(wrong)
+  end
+
+  test "parse2", %{podcast_cre: podcast_cre} do
```
Comment (umurgdk) [inline test/feedme_test.exs]: can you please rename test to something like "parse_podcast_cre"?

File: test/parsers/rss2_test.exs [hunk 2]
```diff
@@ -7,16 +7,18 @@ defmodule Feedme.Test.Parsers.RSS2 do
   setup do
     sample1 = XmlNode.from_file("test/fixtures/rss2/sample1.xml")
     sample2 = XmlNode.from_file("test/fixtures/rss2/sample2.xml")
+    sample3 = XmlNode.from_file("test/fixtures/rss2/cre.xml")
```
Comment (umurgdk) [inline test/parsers/rss2_test.exs]: can you please rename sample name to more descriptive since it's not a standard rss2 feed.

**Comment by umurgdk** [comment]
Hey sorry for my late answer, I was quite busy with daily job.

I was thinking of separating parsers for different kind of feed formats instead of merging them with regular rss2 parser. I have some rough ideas about what can be done. Please give me your opinions.
1. Instead of writing one big `parse` function for each module. We can write each parsed field as separate functions like `parse_author`, `parse_artist`. Then we can separate itunes module to its own. Then we can reuse rss2 module's `parse_xxx` functions in itunes module. But that can end up with some performance issues since every parser try to find right xml node again and again.
2. There can be some kind of extension mechanism. Since feeds like atom, itunes are based on each other. We can think ITunes module as an extension module based on Atom whic is based on RSS2.   Base module can define extension points and call this extension points with already found xml nodes after base data is parsed. That one would perform much better than the first option. But it is harder to implement and needs some architectural design.

**Comment by argl** [comment]
Hi, thanks for the info. I had quite a rough ride with podcast feeds. I stumbled upon one which did not decode because it had some none-utf-8 encoding. xmerl bailed on that. So I tried an implenetation based on Expat, but that did not improve things, after that I lost motivation. For now I am using a hacked expat solution where I pipe the feed through xmllint if there is any parse error and try again.

After some discussions I am now convinced that feeds with non-utf-8 encoding should be ignored anyway. So I halfheartedly planned to rewrite the whole thing again, based on erlsom and a streaming sax parser, My app has to parse tens of thousands of feeds, so memory is of some concern.

That said, on to your questions:
1. That sounds good, modularization would be a good thing. Trading some performance for a much cleaner architecture and extensibility should be a nobrainer.
2. I dont think I understand, itunes is not really based on atom, these are just additional namespaces, basically independent of each other. There are even more namespaces in some feeds which I ignored for now. Can you elaborate how that should work, also how would the resulting structs look like?

As a side note, there are also plans to make a (semantic) feed validator, a thing which can tell the creator of the feed things like "adding chapter marks to your episodes will benefit your users" or "adding contributor information to your epsodes will help people find specific persons". All in the hope that overall feed quality for podcasts will improve.

Another thing I struggled with are "related" feeds like podcasts offering their episodes in different formats and have a feed for each format. The podlove publisher plugin for wordpress generates such feeds. 

And there are also "paged" feeds, where one can page through several feed documents of the same podcast. This was created for feeds with several hundred episodes. Most podcatchers choke on feeds containing that much episodes, so a system to break them up into smaller chunks was created.

Both of these (formats and paged feeds) are porbably outside the scope of feedme, I just wanted to whine a bit;)

So, to sum it up, neither my current setup nor this pull request really satisfy my needs totally, I would love to try a fresh approach. Please let me hear your thoughts.
